server/controllers/complaintHandler.js

```diff
@@ -0,0 +1,118 @@
+const DocComplaint = require('../database/models/docComplaint');
+const UserComplaint = require('../database/models/userComplaint');
+const authentication = ('../authentication');
+
+const createDocComplaint = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    let doccomplaintinst = new DocComplaint({
+      text: req.body.text,
+      fromId: req.body.fromId,
```

Comment:
This should not be retrieved from a post field. Retrieve the user id from the user currently logged in. See [here](https://github.com/chc5/gitgud/blob/master/server/controllers/docHandlers.js#L13)